### PR TITLE
Refs [Task][#439563][POST /account/me/changePassword: Change password for user (include unit test)]

### PR DESCRIPTION
**Pull request by conglt10**

### CODE REVIEW

File: server/routes/me.js [hunk 1]
```diff
@@ -575,4 +576,77 @@ router.post('/avatar', checkJWT, multipartMiddleware, async (req, res) => {
...
+          msg: 'Bad request'
+        });
+      }
+
+      let user = await User.findOne({ username: req.decoded.user.username });
+
+      if (!user) {
+        return res.status(404).json({
+          success: false,
+          msg: 'Account is not exist'
```
Comment (trinhtan) [inline server/routes/me.js]: Account doesn't exist

File: server/routes/me.js [hunk 2]
```diff
@@ -575,4 +576,77 @@ router.post('/avatar', checkJWT, multipartMiddleware, async (req, res) => {
...
+
+      return res.status(200).json({
+        success: true,
+        msg: 'Change password success'
+      });
+    } catch (error) {
+      console.log(error);
+      return res.status(500).json({
+        success: false,
+        msg: 'Change password fail'
```
Comment (trinhtan) [inline server/routes/me.js]: Password change failed

File: server/routes/me.js [hunk 3]
```diff
@@ -575,4 +576,77 @@ router.post('/avatar', checkJWT, multipartMiddleware, async (req, res) => {
...
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    try {
+      const { oldPass, newPass, confirmPass } = req.body;
+
+      if (oldPass === newPass || newPass !== confirmPass) {
```
Comment (dotrungkien) [inline server/routes/me.js]: `oldPass === newPass`
`newPass !== confirmPass`
mỗi cái này là một message khác nhau chứ em?

File: server/routes/me.js [hunk 4]
```diff
@@ -575,4 +576,77 @@ router.post('/avatar', checkJWT, multipartMiddleware, async (req, res) => {
...
+      }
+
+      const SALTROUNDS = 10;
+      const hashVal = await bcrypt.hash(newPass, SALTROUNDS);
+
+      await user.updateOne({ password: hashVal });
+
+      return res.status(200).json({
+        success: true,
+        msg: 'Change password success'
```
Comment (dotrungkien) [inline server/routes/me.js]: Change password successfully

File: server/routes/me.js [hunk 5]
```diff
@@ -575,4 +576,77 @@ router.post('/avatar', checkJWT, multipartMiddleware, async (req, res) => {
...
+      });
+    } catch (error) {
+      console.log(error);
```
Comment (dotrungkien) [inline server/routes/me.js]: bỏ `console.log`

File: server/routes/me.js [hunk 6]
```diff
@@ -575,4 +576,81 @@ router.post('/avatar', checkJWT, multipartMiddleware, async (req, res) => {
...
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    try {
+      const { oldPass, newPass, confirmPass } = req.body;
+
+      if (oldPass === newPass) {
+        return res.status(400).json({
+          success: false,
+          msg: 'New password is not different old password'
```
Comment (dotrungkien) [inline server/routes/me.js]: new password must be different than the old one

File: server/routes/me.js [hunk 7]
```diff
@@ -575,4 +576,81 @@ router.post('/avatar', checkJWT, multipartMiddleware, async (req, res) => {
...
+        });
+      } else if (newPass !== confirmPass) {
+        return res.status(400).json({
+          success: false,
+          msg: 'New Password confirm is not match'
```
Comment (dotrungkien) [inline server/routes/me.js]: confirm password does not match

**Pull request merged**